spec/factorial_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative '../class/solver'
+require 'rspec'
+require 'rspec/mocks'
+
+describe Solver do
+  let(:solver) { Solver.new }
+
+  describe '#factorial' do
+    it 'returns zero if the parameter is zero' do
+      allow(solver).to receive(:factorial).with(0).and_return(0)
+
+      expect(solver.factorial(0)).to eq(0)
```

Comment:
- [ ] When the number is 0, the result should be 1, not 0. Kindly fix this.

Reply:
Suggested changes by the code reviewer have been incorporated. Screen shot attached below for your kind reference.

![image](https://github.com/baqar-abbas/ruby-TDD/assets/117091923/fb4f1dce-4f5c-4468-86c6-6ed3ca24b74e)

Reply:
Send it for a review again